### ISSUE TITLE
[Bug]: Pricing block accessibility issues with crossed items

### ISSUE THREAD

**Issue by bartaz** [canonical/vanilla-framework#5664]
**Describe the bug**

In pricing block pattern the difference between ticked and crossed items is purely visual (by icons and use of muted color). But semantically all items are part of the same list, often labeled as "What's included".

This creates a11y issues for screen readers, because there is no difference for screen reader in how the ticked and crossed items are spelled out, giving impression that everything on the list is included in the package.

**To Reproduce**

1. Go to https://vanillaframework.io/docs/examples/patterns/pricing-block/4-blocks?theme=light
2. Use screen reader to read the items "included" in the pricing packages

**Expected behavior**

There should be some semantic or content difference for the screen readers. For example "not included" could be added as hidden text, or via aria-label on items that are crossed.

**Screenshots**

<img width="795" height="522" alt="Image" src="https://github.com/user-attachments/assets/688739de-6b24-403c-abf2-f7703384d06b" />

**Comment by Safvan-tsy** [comment]
Hi @bartaz , I’d like to work on this issue.

To address this problem, I suggest using the existing `u-off-screen` utility class to make “not included” text available to screen readers while keeping the current visual design.

[#/templates/_macros/vf_pricing-block.jinja#L38-L54](https://github.com/canonical/vanilla-framework/blob/main/templates/_macros/vf_pricing-block.jinja#L38-L54)

like so:
```
    <ul class="p-list--divided">
      {%- for offering in tier.get("tier_offerings") %}
        {%- set item_style = offering.get('list_item_style', 'undefined') -%}
        {% if item_style == 'ticked' -%}
          {% set item_class = "is-ticked" -%}
        {% elif item_style == 'crossed' -%}
          {% set item_class = "is-crossed u-text--muted" -%}
         {% set sr_prefix = '<span class="u-off-screen">Not included </span>' -%}
        {% else -%}
          {% set item_class = "has-bullet" -%}
        {%- endif -%}
        <li class="p-list__item {{ item_class }}">{{ sr_prefix }}{{ offering.list_item_content_html }}</li>
      {% endfor -%}
    </ul>
```
This will allow screen readers to announce “Not included” before,  preserving the current visual layout.
